### PR TITLE
Feature: Restyle Flash Messages with Tailwind.

### PR DESCRIPTION
**Pull request by KevinMulhern**
Because:
* We were using the default Bootstap styles for our flash messages.

This commit:
* Always display flash messages in the upper right in a fixed position so they can be seen if the user is at the bottom of the page.
* Add a flash View Component to handle the different types of flash messages.
* Add a flash Stimulus JS controller to handle dismissing flash messages.
* Cleans up partials used for flash messages - we didn't need two of them.
* Move navbar, announcements and flash partials out of a partial and into the application layout.
* Remove bootstrap helpers we used for selecting the correct styles for our old flash messages.

Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behaviour
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes

Complete the following checkbox ONLY IF it is applicable to your PR. You can complete it later if it is not currently applicable:
-   [x] I have verified all tests and linters pass against my changes, and/or I have included automated tests where applicable

### CODE REVIEW

**Comment by KevinMulhern** [comment]
Before:
Screenshot 2022-08-07 at 13 29 09

After:
Screenshot 2022-08-07 at 13 28 14

File: app/assets/images/icons/x.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
```
Comment (leosoaivan) [inline app/assets/images/icons/x.svg:1]: Is this icon any different from the one we use in our modals?

**Review by leosoaivan — APPROVED** [review]
Just one non-blocking question, otherwise LGTM

File: app/assets/images/icons/x.svg (hunk 1 above)
Comment (KevinMulhern) [reply app/assets/images/icons/x.svg:1]: I think we're using font awesome for the modal close buttons @leosoaivan. I'll create a chore to replace it, we're using these [Heroicons](https://heroicons.com/) in more places than FA now 😆

**Comment by KevinMulhern** [comment]
I've updated the styling to be a bit more subtle
Screenshot 2022-09-03 at 19 56 19

**Pull request merged**